Add logical and bitwise operators for Java

jeff303 opened the pull request:
## What GitHub issue does this PR apply to?

Resolves #620


## What changed and why?

Added logical operators that are supported in Java.  The "and assignment" variants are not implemented in Java (see [here](https://stackoverflow.com/q/2324549)).

## Checklist

- [X] I claimed any associated issue(s) and they are not someone else's
- [X] I have looked at documentation to ensure I made any revisions correctly
- [X] I tested my changes locally to ensure they work
- [ ] (If editing Django) I have added or edited any appropriate unit tests for my changes

jeff303 committed:
Add logical operators for Java

geekygirlsarah committed:
Merge branch 'main' into java-add-logical-operators

gutogalego reviewed (approved):
This looks good. There are still a couple of missing bitwise operators, [reference here](https://www.geeksforgeeks.org/bitwise-operators-in-java/), although I don't know when they were introduced.

geekygirlsarah reviewed (changes requested):
Like @gutogalego said, there's some missing operators. (I know you made the issue originally for just logic operators but it really makes sense to make the PR update all the operators, not just some of them.)

Also it'd be helpful to show where the operands go relative to operators. (Most languages do like `a + b` but some like Lisp do `(+ a b)` so I'm trying to get those updated.)

If you can add those in, I can merge this in.

![image](https://user-images.githubusercontent.com/2601974/197349906-d8f882c2-97bb-428e-8088-153435588790.png)

Commit:
PR feedback

For Java 11

Adding more expanded code sections for bitwise operators (so we can see
the placement of args)

Adding not-implemented sections where missing

jeff303 commented:
> Like @gutogalego said, there's some missing operators. (I know you made the issue originally for just logic operators but it really makes sense to make the PR update all the operators, not just some of them.)
> 
> Also it'd be helpful to show where the operands go relative to operators. (Most languages do like `a + b` but some like Lisp do `(+ a b)` so I'm trying to get those updated.)
> 
> If you can add those in, I can merge this in.

I updated the Java 11 language file with all of these suggestions.  Once this looks good, I can apply the same changes to the other Java versions (since all of this will be identical across all three).

geekygirlsarah commented on web/thesauruses/java/15/operators.json:
```diff
@@ -106,6 +106,24 @@
       "not-implemented": "true",
       "name": "Is not operator"
     },
+    "logical_and": {
```
You added operand placement in Java 11's file but not 15 and 17. Could you do the same for those as well?

geekygirlsarah reviewed (changes requested):
Make similar changes to v15 and v17 and this should be good to merge!

geekygirlsarah committed:
Merge branch 'main' into java-add-logical-operators

geekygirlsarah commented on web/thesauruses/java/11/operators.json:
```diff
@@ -31,27 +31,48 @@
     "increment": {
-      "code": "++",
+      "comment": "Supports both pre and post increment",
+      "code": "x++\n++x",
```
Your examples are post-increment then pre-increment, but mention it's pre then post. Can you swap one of those so they match?

geekygirlsarah commented on web/thesauruses/java/11/operators.json:
```diff
@@ -31,27 +31,48 @@
     "decrement": {
-      "code": "--",
+      "comment": "Supports both pre and post decrement",
+      "code": "x--\n--x",
```
Same here.

geekygirlsarah reviewed (changes requested):
Two tiny swaps and I'll merge it!

jeff303 committed:
Fix pre-and post increment order

jeff303 committed:
Applying changes to Java 15 and 17 files

jeff303 committed:
Operand placement for logical ops

jeff303 commented:
> Two tiny swaps and I'll merge it!

Finally got these all fixed up (I believe).  Tested the comparison page between 11-15 and 11-17.

geekygirlsarah committed:
Merge branch 'main' into java-add-logical-operators

geekygirlsarah reviewed (approved):
Great! I'll merge this in. Thanks for the changes!
